src/vdom/index.js

```diff
@@ -10,19 +12,38 @@ export function isSameNodeType(node, vnode, hydrating) {
 	if (typeof vnode==='string' || typeof vnode==='number') {
 		return node.splitText!==undefined;
 	}
-	if (typeof vnode.nodeName==='string') {
-		return !node._componentConstructor && isNamedNode(node, vnode.nodeName);
+	const { _componentConstructor } = node;
+	const { nodeName } = vnode;
+	if (typeof nodeName==='string') {
```

Comment:
I think you actually want the custom element check here, otherwise all custom elements are treated as the same element during hydration, or always as a mismatch.